codec.go

```diff
@@ -39,19 +39,25 @@ func (e *DecodeProtocolMessageFieldError) Error() string {
 // command, etc cannot be cast, returns DecodeProtocolMessageFieldError.
 // See also godoc for json.Unmarshal, which is used for underlying decoding.
 func DecodeProtocolMessage(data []byte) (Message, error) {
-	var protomsg ProtocolMessage
-	if err := json.Unmarshal(data, &protomsg); err != nil {
+	var m struct {
+		Seq     int    `json:"seq"`
```

Comment:
Consider embedding the ProtocolMessage type and we can keep the m.GetSeq() call on line 60

Reply:
Embedding `ProtocolMessage` makes sense to me. However, I don't see the advantage of using the `GetSeq()` accessor method. The field is public and we can work with the concrete type and not the interface. Also, getters aren't encouraged by the style guide https://google.github.io/styleguide/go/decisions#getters.

Reply:
Not sure the original motivation for GetSeq, I was mostly suggesting to stay consistent with how we are accessing Seq throughout the package. Using Seq directly certainly seems fine here.